File: src/SelfTeacher/Web/SelfTeacher.Web/ClientApp/src/services/interfaces/iuser.service.ts
import { UserDto } from "../../models/UserDto";
import { ListDto } from "../../models/ListDto";

export abstract class IUserService {

    /**
     * login method user service
     * @param username - login of user
     * @param password - password of user
     * @returns Promise after request and adding to local storage
     */
    public abstract login(username: string, password: string): Promise<any>;

    /**
     * Logout function for remove user info from local storage
     */
    public abstract logout(): void;

    /**
     * Register user in system by user info
     */
    public abstract register(user: UserDto): Promise<any>;

    /**
     * Method for getting all users in system
     * @returns Promise with all users in system
     */
    public abstract getAll(): Promise<ListDto<UserDto>>;

    /**
     * Method for getting user by ID
     * @param id - id of user
     * @returns Promise with User information
     */
    public abstract getById(id: string): Promise<UserDto>;

    /**
     * Method for update user information
     * @param User - UserDto for update
     * @returns Promise request for update
     */
    public abstract update(user: UserDto): Promise<UserDto>;

    /**
     * 
     * @param id - id of user for deleting
     * @returns Promise request of deleting user
     */
    public abstract delete(id: string): Promise<void>;

    /**
     * @returns Flag if user in login
     */
    public abstract IsLoginUser(): boolean;
}

File: src/SelfTeacher/Service/ServiceTeacher.Service.Domain/Services/IUserService.cs
using ServiceTeacher.Service.Domain.Entities;
using System;
using System.Collections.Generic;

namespace ServiceTeacher.Service.Domain.Services
{
    /// <summary>
    /// Interface for work with users
    /// </summary>
    public interface IUserService
    {
        /// <summary>
        /// Method for authenricate user
        /// </summary>
        /// <param name="username"></param>
        /// <param name="password"></param>
        /// <returns></returns>
        object Authenticate(string username, string password);

        /// <summary>
        /// Mehtod for authentincation by VK API
        /// </summary>
        /// <param name="accessToken">Access token from API</param>
        /// <returns></returns>
        object AuthenticateByAccessTokenVk(string accessToken);

        /// <summary>
        /// Method for getting all users
        /// </summary>
        /// <returns></returns>
        IEnumerable<User> GetAll();

        /// <summary>
        /// Confirmation user method
        /// </summary>
        /// <param name="userId">Identificator of user</param>
        /// <param name="code">Confirmation code</param>
        void ConfirmUser(Guid userId, string code);

        /// <summary>
        /// Method for taking one user by ID
        /// </summary>
        /// <param name="id"></param>
        /// <returns></returns>
        User GetById(Guid id);

        /// <summary>
        /// Registration of users
        /// </summary>
        /// <param name="user"></param>
        /// <param name="password"></param>
        /// <returns></returns>
        void Create(User user, string password);

        /// <summary>
        /// Registration Vk users
        /// </summary>
        /// <param name="user"></param>
        string CreateVkUser(User user);

        /// <summary>
        /// Edit information for user
        /// </summary>
        /// <param name="user"></param>
        /// <param name="password"></param>
        void Update(User user, string password = null);
    }
}
